### DISCUSSION TITLE
Error on axisBottom format when updating data

### DISCUSSION THREAD

**Discussion by dmendiboure** [plouc/nivo#2594] [Q&A]
Hi there,

Here's the problem I'm having, I'm receiving data from an API with several tables (each for a defined period). I have buttons that allow to change from one period to another and a function that allows to format the x-axis in order to display the label differently depending on the period.

However, when the state is changed, the function for formatting the axis is called up with the old values, causing the application to crash.

Here's an example: https://codesandbox.io/p/sandbox/tender-chatterjee-j4pt7h?file=%2Fsrc%2FApp.tsx%3A32%2C17

**Comment by plouc** [comment]
Seems like an edge case, there must be something wrong with how the updates are chained, there's a lot of memoization internally, could be a missing dependency in one of the hooks.
